base/uk.ac.stfc.isis.ibex.e4.ui.perspectiveswitcher/src/uk/ac/stfc/isis/ibex/e4/ui/perspectiveswitcher/PerspectivesProvider.java

```diff
@@ -125,14 +133,74 @@ public MPerspectiveStack getTopLevelStack() {
      * Switches to a given perspective.
-     * @param perspective the the perspective to switch to
+     * @param perspective the perspective to switch to
      */
     public void switchPerspective(MPerspective perspective) {
-    	partService.switchPerspective(perspective);
+        checkIfDaeIsCurrentPerspective(perspective);
+    }
+
+    private void checkIfDaeIsCurrentPerspective(MPerspective perspectiveToLoad) {
+        for (MPerspective perspectiveInService : modelService.findElements(app, null, MPerspective.class, null)) {
+            if (isSelected(perspectiveInService)) {
+                if (perspectiveInService.getLabel().equals("DAE")) {
+                    daeHasBeenChanged(perspectiveInService, perspectiveToLoad);
+                } else {
+                    partService.switchPerspective(perspectiveToLoad);
+                    break;
+                }
+            }
+        }
+    }
+
+    private void daeHasBeenChanged(MPerspective perspectiveInService, MPerspective perspectiveToLoad) {
+        if(daeViewModel.getIsChanged()) {
+            dialogConfirmed(perspectiveInService, perspectiveToLoad);
+        } else {
+            partService.switchPerspective(perspectiveToLoad);
+        }
+    }
+
+    private void dialogConfirmed(MPerspective perspectiveInService, MPerspective perspectiveToLoad){
+        if (leaveWithoutApplyingDaeChangesDialog(perspectiveInService.getLabel(), shell)) {
+            partService.switchPerspective(perspectiveToLoad);
+        } 
+    }
+
+    private void checkIfDaeIsCurrentPerspective(String id) {
+        for (MPerspective perspectiveInService : modelService.findElements(app, null, MPerspective.class, null)) {
+            if (isSelected(perspectiveInService)) {
+                if (perspectiveInService.getLabel().equals("DAE")) {
+                    daeHasBeenChanged(perspectiveInService, id);
+                } else {
+                    partService.switchPerspective(id);
+                    break;
+                }
+            }
+        }
+    }
+
+    private void daeHasBeenChanged(MPerspective perspectiveInService, String id) {
+        if(daeViewModel.getIsChanged()) {
+            dialogConfirmed(perspectiveInService, id);
+        } else {
+            partService.switchPerspective(id);
+        }
+    }
+
+    private void dialogConfirmed(MPerspective perspectiveInService, String id){
+        if (leaveWithoutApplyingDaeChangesDialog(perspectiveInService.getLabel(), shell)) {
+            partService.switchPerspective(id);
+        }
+    }
+
+    private boolean leaveWithoutApplyingDaeChangesDialog(String perspectiveName, Shell shell) {
+        return MessageDialog.openQuestion(shell, "Confirm Leaving " + perspectiveName + " Perspective",
+                "You have unapplied changes in the \"Experiment Setup\" tab, are you sure you want to leave "
+                + "the " + perspectiveName + " perspective? \n\nThese changes are highlighted for you to find.");
```

Comment:
I think it would be better to say:

"You have unapplied changes in the Experiment Setup tab, are you sure you want to leave the <perspectiveName> perspective? These changes will remain here but will not be applied"

Unless we want to discard the changes when leaving the perspective? Currently they all remain in place for when you navigare back.